datafusion/physical-expr-common/src/physical_expr.rs

```diff
@@ -31,8 +31,27 @@ use datafusion_expr_common::columnar_value::ColumnarValue;
 use datafusion_expr_common::interval_arithmetic::Interval;
 use datafusion_expr_common::sort_properties::ExprProperties;
 
-/// See [create_physical_expr](https://docs.rs/datafusion/latest/datafusion/physical_expr/fn.create_physical_expr.html)
-/// for examples of creating `PhysicalExpr` from `Expr`
+/// [`PhysicalExpr`]s represent expressions such as `A + 1` or `CAST(c1 AS int)`.
```

Comment:
it was strange there was no explanation of what a `PhysicalExpr` was on the trait, so let's fix that (moved the docs from `create_physical_expr` and revamped it a bit)